File: studentassistant/src/main/java/ru/erdenian/studentassistant/activity/SplashActivity.java
package ru.erdenian.studentassistant.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import ru.erdenian.studentassistant.ulils.FileUtils;

/**
 * Activity, открывающееся при запуске приложения.
 * Показывает картинку на весь экран, пока подгружается следующее Activity.
 *
 * @author Ilya Solovyev
 * @version 0.0.0
 * @since 0.0.0
 */
public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        FileUtils.initialize(this);

        /*ImmutableSortedSet<Lesson> lessons = ImmutableSortedSet.of(
                new Lesson("Конструирование ПО", "Лабораторная работа",
                        ImmutableSortedSet.of("Федоров Алексей Роальдович", "Федоров Петр Алексеевич"), ImmutableSortedSet.of("4212а"),
                        new LocalTime(14, 20), new LocalTime(15, 50),
                        Lesson.RepeatType.BY_WEEKDAY, 5, ImmutableList.of(false, true), null),
                new Lesson("Конструирование ПО", "Лабораторная работа",
                        ImmutableSortedSet.of("Федоров Алексей Роальдович"), ImmutableSortedSet.of("4212а"),
                        new LocalTime(18, 10), new LocalTime(19, 40),
                        Lesson.RepeatType.BY_WEEKDAY, 5, ImmutableList.of(false, true), null));

        ImmutableSortedSet<Semester> semesters = ImmutableSortedSet.of(
                new Semester("Семестр 5", new LocalDate(2016, 9, 1), new LocalDate(2016, 12, 31),
                        lessons, ImmutableSortedSet.<Homework>of()),
                new Semester("Семестр 6", new LocalDate(2017, 9, 1), new LocalDate(2017, 12, 31),
                        lessons, ImmutableSortedSet.<Homework>of()));

        ScheduleManager.setSemesters(semesters);*/

        Intent intent = new Intent(this, ScheduleActivity.class);
        startActivity(intent);
        finish();
    }
}

File: studentassistant/src/main/kotlin/ru/erdenian/studentassistant/activity/SplashActivity.kt
package ru.erdenian.studentassistant.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.fatboyindustrial.gsonjodatime.Converters
import com.google.common.collect.ImmutableList
import com.google.common.collect.ImmutableSortedSet
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import org.jetbrains.anko.defaultSharedPreferences
import org.jetbrains.anko.startActivity
import org.jetbrains.anko.startService
import ru.erdenian.studentassistant.netty.*
import ru.erdenian.studentassistant.schedule.Lesson
import ru.erdenian.studentassistant.schedule.LessonRepeat
import ru.erdenian.studentassistant.schedule.ScheduleManager
import ru.erdenian.studentassistant.schedule.Semester
import ru.erdenian.studentassistant.service.ScheduleService

class SplashActivity : AppCompatActivity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    ScheduleManager.initialize(applicationContext)

    defaultSharedPreferences.apply {
      val login = getString("login", "null")
      val password = getString("password", "null")

      nettyQuery("$login;$password::getuser::") {
        val user = Gson().fromJson(it, User::class.java)

        nettyQuery("$login;$password::getschedule::${user.groupId}") {
          val args = it.split(';')
          val semesterJson = args[0]

          val gson = Converters.registerAll(GsonBuilder())
              .registerTypeAdapter(ImmutableSortedSet::class.java, ImmutableSortedSetDeserializer())
              .registerTypeAdapter(ImmutableList::class.java, ImmutableListDeserializer())
              .registerTypeAdapter(LessonRepeat::class.java, LessonRepeatDeserializer())
              .create()

          val semester = gson.fromJson(semesterJson, Semester::class.java)
          val semester1 = Semester(semester.name, semester.firstDay, semester.lastDay, semester.id)
          ScheduleManager.removeSemester(semester1.id)
          ScheduleManager.addSemester(semester1)

          for (i in 1..(args.size - 1)) {
            ScheduleManager.addLesson(semester.id, gson.fromJson(args[i], Lesson::class.java))
          }
        }
      }
    }

    startService<ScheduleService>()
    startActivity<ScheduleActivity>()
    finish()
  }
}
